src/compiler/sys/in-memory-fs.ts

```diff
@@ -1,77 +1,226 @@
   const outputTargetTypes = new Map<string, string>();
 
-  const accessData = async (filePath: string) => {
+  /**
+   * Check if a file exists at a provided path. This function will attempt to
+   * use the in-memory cache before performing a blocking read. In the event of
+   * a cache hit, the content from the cache will be returned and the read skipped.
+   *
+   * @param filePath the path to the file to read
+   * @returns `true` if the file exists, `false` otherwise
+   */
+  const access = async (filePath: string): Promise<boolean> => {
     const item = getItem(filePath);
-    if (typeof item.exists === 'boolean') {
-      return {
-        exists: item.exists,
-        isDirectory: item.isDirectory,
-        isFile: item.isFile,
-      };
-    }
-
-    const data = {
-      exists: false,
-      isDirectory: false,
-      isFile: false,
-    };
 
-    const s = await stat(filePath);
-    if (s) {
-      item.exists = s.exists;
-      item.isDirectory = s.isDirectory;
-      item.isFile = s.isFile;
-
-      data.exists = item.exists;
-      data.isDirectory = item.isDirectory;
-      data.isFile = item.isFile;
-    } else {
-      item.exists = false;
+    if (typeof item.exists !== 'boolean') {
+      const stats = await stat(filePath);
+      return stats.exists;
     }
-
-    return data;
-  };
-
-  const access = async (filePath: string) => {
-    const data = await accessData(filePath);
-    return data.exists;
+    return item.exists;
   };
 
   /**
    * **Synchronous!!! Do not use!!!**
    * (Only typescript transpiling is allowed to use)
    *
-   * Synchronously get information about a file from a provided path. This function will attempt to use an in-memory
-   * cache before performing a blocking read.
+   * Synchronously get information about a file from a provided path. This
+   * function will attempt to use an in-memory cache before performing a
+   * blocking read.
    *
-   * In the event of a cache hit, the content from the cache will be returned and skip the read.
+   * In the event of a cache hit, the content from the cache will be returned
+   * and skip the read.
    *
    * @param filePath the path to the file to read
    * @returns `true` if the file exists, `false` otherwise
    */
   const accessSync = (filePath: string): boolean => {
     const item = getItem(filePath);
     if (typeof item.exists !== 'boolean') {
-      const s = statSync(filePath);
-      item.exists = s.exists;
-      item.isDirectory = s.isDirectory;
-      item.isFile = s.isFile;
+      const stats = statSync(filePath);
+      return stats.exists;
     }
     return item.exists;
   };
 
-  const copyFile = async (src: string, dest: string) => {
+  /**
+   * Copy a file from `src` to `dest`. Note that this merely queues the file
+   * for copying, the copy isn't actually committed.
+   *
+   * @param src the path to the source file
+   * @param dest the destination the source file should be copied to
+   * @returns a promise
```

Comment:
No action required - IMO we can elide `@returns` for empty Promises from JSDoc. The linter is configured to permit that, but I don't believe we could configure it to enforce consistency (I.E. "warn us if we add `@returns` on a fn that returns `void` or `Promise<void>`), hence the no change needed here